plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/RewriteStringComparison.java

```diff
@@ -0,0 +1,93 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.oracle;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.QueryParameter;
+import io.trino.plugin.jdbc.expression.ComparisonOperator;
+import io.trino.plugin.jdbc.expression.ParameterizedExpression;
+import io.trino.spi.expression.Call;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.VarcharType;
+import oracle.jdbc.OracleTypes;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argument;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argumentCount;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.call;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.functionName;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+
+public class RewriteStringComparison
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> FIRST_ARGUMENT = newCapture();
+    private static final Capture<Variable> SECOND_ARGUMENT = newCapture();
+    private static final Pattern<Call> PATTERN = call()
+            .with(type().equalTo(BOOLEAN))
+            .with(functionName().matching(Stream.of(ComparisonOperator.values())
+                    .filter(comparison -> comparison != ComparisonOperator.IS_DISTINCT_FROM)
+                    .map(ComparisonOperator::getFunctionName)
+                    .collect(toImmutableSet())
+                    ::contains))
+            .with(argumentCount().equalTo(2))
+            .with(argument(0).matching(variable().with(type().matching(type -> type instanceof CharType || type instanceof VarcharType)).capturedAs(FIRST_ARGUMENT)))
+            .with(argument(1).matching(variable().with(type().matching(type -> type instanceof CharType || type instanceof VarcharType)).capturedAs(SECOND_ARGUMENT)));
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        ComparisonOperator comparison = ComparisonOperator.forFunctionName(expression.getFunctionName());
+        Variable firstArgument = captures.get(FIRST_ARGUMENT);
```

Comment:
nit: operand

Reply:
same as in https://github.com/trinodb/trino/pull/19996#discussion_r1447147979